src/ethereum/utils.py

```diff
@@ -59,3 +76,354 @@ def ceil32(value: Uint) -> Uint:
         return value
     else:
         return value + ceiling - remainder
+
+
+def to_valid_address(data: bytes) -> Address:
```

Comment:
Can we maybe go with something like:

```python
def to_address(data: Union[U256, UInt]) -> Address:
```

Or are there places where an actual `bytes` or `bytearray` is passed in?

If we can use the integer types, we can simplify this function to use `to_be_bytes32`:

```python
def to_address(data: Union[U256, UInt]) -> Address:
    return data.to_be_bytes32()[-20:]
```

Reply:
+1